app/services/container_dashboard_service.rb

```diff
@@ -1,13 +1,16 @@
 class ContainerDashboardService
-  def initialize(provider_id)
+  def initialize(provider_id, controller)
```

Comment:
if the controller is passed only to extract the user, why not pass just the user? (I didn't notice other usages of the controller in the service)